src/accel/detail/RngReseed.cu

```diff
@@ -0,0 +1,73 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/detail/RngReseed.cu
+//---------------------------------------------------------------------------//
+#include "RngReseed.hh"
+
+#include "corecel/device_runtime_api.h"
+#include "corecel/Assert.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/KernelParamCalculator.device.hh"
+#include "celeritas/random/RngEngine.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+// KERNELS
+//---------------------------------------------------------------------------//
+/*!
+ * Reinitialize the RNG states on device using the Geant4 Event ID.
+ */
+__global__ void reseed_rng_kernel(DeviceCRef<RngParamsData> const params,
+                                  DeviceRef<RngStateData> const state,
+                                  size_type event_id)
+{
+    auto tid = TrackSlotId{
+        celeritas::KernelParamCalculator::thread_id().unchecked_get()};
+    if (tid.get() < state.size())
+    {
+        TrackSlotId tsid{tid.unchecked_get()};
+        RngEngine::Initializer_t init;
+#if CELERITAS_CORE_RNG == CELERITAS_CORE_RNG_XORWOW
+        init.seed = params.seed[0];
+#else
+        init.seed = params.seed;
+#endif
```

Comment:
Maybe the RNG engine initializer should have a `seed` that's the same type as `params.seed`?

Reply:
Yeah good point, I can change the initializer seed to an array. Is this still a plan for xorwow?
```
// TODO: 256-bit seed used to generate initial states for the RNGs
// For now, just 4 bytes (same as our existing cuda/hip interface)
Array<uint_t, 1> seed;
```

Reply:
It could be? 😅 it would make sense to do, but yeah I don't see any practical use until we start feeding hashed time stamps from a python front end